File: src/test/java/io/prometheus/client/TestableTimeProvider.java
package io.prometheus.client;

@SuppressWarnings("unused")
public class TestableTimeProvider extends SimpleTimer.TimeProvider {

    private long baseTime = System.currentTimeMillis();

    public static void install() {
        SimpleTimer.defaultTimeProvider = new TestableTimeProvider();
    }

    @Override
    long nanoTime() {
        final long curr = baseTime;
        baseTime += 1979L;
        return curr;
    }
}

File: src/test/java/io/prometheus/client/TestableTimeProvider.kt
package io.prometheus.client

internal class TestableTimeProvider : SimpleTimer.TimeProvider() {
    private var baseTime = System.currentTimeMillis()

    public override fun nanoTime(): Long {
        val curr = baseTime
        baseTime += 1979L
        return curr
    }

    companion object {
        fun install() {
            SimpleTimer.defaultTimeProvider = TestableTimeProvider()
        }
    }
}